production/ksonnet/loki/ingester.libsonnet

```diff
@@ -5,6 +5,28 @@ local k = import 'ksonnet-util/kausal.libsonnet';
   local volumeMount = k.core.v1.volumeMount,
   local statefulSet = k.apps.v1.statefulSet,
 
+  // The ingesters should persist TSDB blocks and WAL on a persistent
+  // volume in order to be crash resilient.
+  local ingester_data_pvc =
+    pvc.new() +
+    pvc.mixin.spec.resources.withRequests({ storage: $._config.ingester_data_disk_size }) +
+    pvc.mixin.spec.withAccessModes(['ReadWriteOnce']) +
+    pvc.mixin.spec.withStorageClassName($._config.ingester_data_disk_class) +
+    pvc.mixin.metadata.withName('ingester-data'),
+
+  newIngesterStatefulSet(name, container, with_anti_affinity=true)::
```

Comment:
Q: why is there leading `::`? is it needed?. Also comments following can be removed as well?

Reply:
trailing `::`, and because it's a function so we don't want it to be rendered as json

some of the commented code I still intend to port more of (via this PR currently, but if that changes I will remove the comments and do so via new PRs), I should have left a PR note about that earlier :+1: